samples/net/mqtt/src/modules/transport/client_id/client_id.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/random/rand32.h>
+#include <hw_id.h>
+#include <stdio.h>
+#include <string.h>
+
+int client_id_get(char *buffer, size_t buffer_size)
+{
+	int err;
+
+	if (strlen(CONFIG_MQTT_SAMPLE_TRANSPORT_CLIENT_ID) > 0) {
+		snprintk(buffer, buffer_size, "%s", CONFIG_MQTT_SAMPLE_TRANSPORT_CLIENT_ID);
```

Comment:
We're potentially cutting the client ID short here, which may cause issues with TLS

Reply:
True, ill add a check for the retval from `snprintk` and return an error if this occurs.